### ISSUE TITLE
false positive of lacking impl, if given from `derive_more` crate

### ISSUE THREAD

**Issue by KisaragiEffective** [intellij-rust/intellij-rust#9216]
## Environment

* **IntelliJ Rust plugin version:** 0.4.176.4815-222
* **Rust toolchain version:** 1.62.0 (a8314ef7d 2022-06-27) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2022.2.1 Ultimate Edition (IU-222.3739.54)
* **Operating system:** Linux 5.10.0-16-amd64
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Code can be compiled (and exit with 0), but false diagnostic is generated.

## Steps to reproduce
A minimal case is shown below:

```rs
use derive_more::Display;

#[derive(Display)]
struct X;

fn main() {
    println!("{}", X);
//                 ^
//                 `X` doesn't implement `Display` (required by {})
}
```

```toml
[package]
name = "intellij-rust-false-positive-display-reproduction"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
derive_more = "0.99.17"
```

**Comment by Undin** [comment]
@KisaragiEffective try to enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) features
See #6908 for more details

**Comment by Khoulaiz** [comment]
I had a similar issue (derived Display as well for enum Error). Enabling the org.rust.macros.proc feature resolved the problem.
